image-process - Added functionality to process files based on extensions jpg, png, jpeg

ashishjullia opened the pull request:
Adds the following functionality:

- Process files based on file extensions instead of mime type.
- Convert images with extensions jpg/jpeg/png to pdf and then perform ocr

ashishjullia committed:
image-process - Added functionality to process files based on extensions jpg, png, jpeg

woozyking commented on app.py:
```diff
@@ -42,19 +41,31 @@ def ocrmypdf_api():
+    if file_extension not in allowed_extensions:
+        return "Invalid file type, please upload a PDF or image file", 400
+
+    if file_extension in {".jpg", ".jpeg", ".png"}:
+    # convert the image to PDF
+        with Image.open(uploaded_file) as img:
+            with tempfile.NamedTemporaryFile(suffix=".pdf", delete=False) as temp_file:
+                img.convert("RGB").save(temp_file, "PDF")
+            pdf_file = open(temp_file.name, 'rb').read()
+    else:
+        pdf_file = uploaded_file.read() 
```
`ocrmypdf` supports image natively, I would try through that route before resorting to pillow/PIL

woozyking commented on app.py:
```diff
@@ -42,19 +41,31 @@ def ocrmypdf_api():
+    if file_extension in {".jpg", ".jpeg", ".png"}:
+    # convert the image to PDF
+        with Image.open(uploaded_file) as img:
+            with tempfile.NamedTemporaryFile(suffix=".pdf", delete=False) as temp_file:
```
`NamedTemporaryFile` has been imported already, no need to import the module again and have a mixed usage pattern

woozyking commented on app.py:
```diff
@@ -42,19 +41,31 @@ def ocrmypdf_api():
+        return "Invalid file type, please upload a PDF or image file", 400
+
+    if file_extension in {".jpg", ".jpeg", ".png"}:
+    # convert the image to PDF
```
indention of the comment is a bit odd

woozyking commented on app.py:
```diff
@@ -1,11 +1,11 @@
 import ocrmypdf
-
+from PIL import Image
+import tempfile
```
1. I would group the import of standard libs together and keep the 3rd party below
2. `from tempfile import NamedTemporaryFile` already exists

woozyking commented on app.py:
```diff
@@ -42,19 +41,31 @@ def ocrmypdf_api():
-    with NamedTemporaryFile(suffix=filename_parts[1]) as output_file, BytesIO(
-        uploaded_file.read()
-    ) as input_buffer:
+    with NamedTemporaryFile(suffix=".pdf") as pdf_output_file, NamedTemporaryFile(suffix=".pdf") as output_file:
```
`BytesIO` was used specifically to minimize disk I/O

ashishjullia committed:
feat(api): Fixes issues with NamedTemporaryFile/tempfile, avoid of use
- Uses BytesIO in favor of minimizing disk I/O
- Used ocrmypdf bult-in ocr image conversion/read capabilities

woozyking commented on app.py:
```diff
@@ -42,20 +37,29 @@ def ocrmypdf_api():
-    with NamedTemporaryFile(suffix=filename_parts[1]) as output_file, BytesIO(
-        uploaded_file.read()
-    ) as input_buffer:
+    with BytesIO(pdf_file) as pdf_output_file, BytesIO() as output_file:
```
there's a reason for one of the tempfile usage here to workaround RAM limit when the running instance is highly constraint on that resource
